### PR TITLE
Task 384-385 Stance Detection

### PR DESCRIPTION
**Pull request by cosmicishan**
task384_Stance_Detection_text_generation -> Given a tweet, classify it into one of these categories: Atheism, Climate Change is a Real Concern, Feminist Movement, Hillary Clinton, and Legalization of Abortion. 

task385_Stance_Detection_text_generation -> Given a tweet and type of that tweet, generate the stance of the tweet. 

Please Review

### CODE REVIEW

**Comment by danyaljj** [comment]
@XudongOliverShen wondering if you can help with this PR's review.

File: tasks/task384_stance_detection_text_generation.json [hunk 1]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "In the absence of God, What are you? You are nothing more than an accident or a re-arrangement of molecules. #bible #SemST",
+            "output": [
+                "Atheism"
+            ]
+        },
+        {
+            "input": "You better go get my ball.... #SemST",
+            "output": [
+                "Atheism"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: why is this sentence labeled atheism?

File: tasks/task384_stance_detection_text_generation.json [hunk 2]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "Minister @manupulgarvidal about the importance of #INDCs to build momentum to #Paris2015 at press conference at #UN #SemST",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Why Is The Pope Upset?  via @HuffPostComedy #UnzippedTruth #PopeFrancis #SemST",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: why is this sentence labeled climate change?

File: tasks/task384_stance_detection_text_generation.json [hunk 3]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "Coldest and wettest summer in memory. #mosquitoes #mowtwiceaweek #brrrr #SemST",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Pope Francis really makin the rounds  #PopeFrancis #SemST",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: reason for the label climate change?

File: tasks/task384_stance_detection_text_generation.json [hunk 4]
```diff
@@ -0,0 +1,16910 @@
...
+        },
+        {
+            "input": "Men that want \"credit\" from women for being feminists are not feminists at all.  They're just men wanting more privileges #vaw #p2 #SemST",
+            "output": [
+                "Feminist Movement"
+            ]
+        },
+        {
+            "input": "@oakleyftgaga you should be nice to me anyways #NicerInternet #SemST",
+
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: the sentence seems not related to Hillary Clinton

Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: seems not necessarily related to Hillary

Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: reasons for related to Hillary?

Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: reasons for related to Hillary?

Comment (XudongOliverShen) [inline tasks/task384_stance_detection_text_generation.json]: reasons for related to legalization of abortion?

File: tasks/task385_stance_detection_text_generation.json [hunk 5]
```diff
@@ -0,0 +1,16910 @@
...
+        },
+        {
+            "input": "I believe that too often religious people do good for the same reasons non-religious do, but they give undo credit to their faith. #SemST",
+            "output": "FAVOUR, AGAINST",
+            "explanation": "The output only supposed to be one label but it generate 2 labels. "
+        },
+        {
+            "input": "If you say climate change is a myth, you may want to become a fantasy author since you already live in an alternate reality. ",
+            "output": "Climate Change is a Real Concern",
+            "explanation": "The output supposed to generate the stance of the tweet and not the target it gives."
```
Comment (XudongOliverShen) [inline tasks/task385_stance_detection_text_generation.json]: type is missing in both positive and negative examples

File: tasks/task385_stance_detection_text_generation.json [hunk 6]
```diff
@@ -0,0 +1,16910 @@
...
+            "output": [
+                "FAVOR"
+            ]
+        },
+        {
+            "input": "Type: Feminist Movement \n Sentence: .@feminismvibes Men don't admit to being raped as readily as women do. So your question is feminist bullshit. #SemST",
+            "output": [
+                "AGAINST"
+            ]
+
```
Comment (XudongOliverShen) [inline tasks/task385_stance_detection_text_generation.json]: I am in doubt if the sentence is related to feminist movement.

Comment (XudongOliverShen) [inline tasks/task385_stance_detection_text_generation.json]: not very clear the sentence is in favor of feminist movement

**Review by XudongOliverShen — COMMENTED** [review]
1. shuffle instances
2. rename task 384 and 385 so to distinguish them
3. why is there a "#SemST" at the end of every sentence? is it important? could it be removed since it's there in every sentence?
4. the association between the sentences and the labels seems very vague (see below comments). I briefly went through the instances and find many sentences that I think do not have a clear association with the label. Could you either explain why are these sentences labeled this way or double check to remove these sentences?
5. type is missing in positive and negatice examples in task 385.

**Comment by cosmicishan** [comment]
> 1. shuffle instances
> 2. rename task 384 and 385 so to distinguish them
> 3. why is there a "#SemST" at the end of every sentence? is it important? could it be removed since it's there in every sentence?
> 4. the association between the sentences and the labels seems very vague (see below comments). I briefly went through the instances and find many sentences that I think do not have a clear association with the label. Could you either explain why are these sentences labeled this way or double check to remove these sentences?

That's how the Dataset is given. I get the link of paper and it's dataset from the google sheet shared with all of us and I provided link in json file. Sure some target and stance may not make sense but I am not the one who annotated the data. I just make it from what it was given to me.

**Comment by XudongOliverShen** [comment]
Hi Ishan. Yes, I noticed these artifacts are from the dataset itself. I just think, if you have the time, you could do some pruning to get rid of these illegitimate instances. But I also understand it might be too taxing. If so, we can discuss other alternatives.

Just my proposal. : )

**Comment by cosmicishan** [comment]
> Hi Ishan. Yes, I noticed these artifacts are from the dataset itself. I just think, if you have the time, you could do some pruning to get rid of these illegitimate instances. But I also understand it might be too taxing. If so, we can discuss other alternatives.
> 
> Just my proposal. : )

Sure I can do that. Please point out other mistakes as well so I can fix it all up.

**Comment by cosmicishan** [comment]
> Hi Ishan. Yes, I noticed these artifacts are from the dataset itself. I just think, if you have the time, you could do some pruning to get rid of these illegitimate instances. But I also understand it might be too taxing. If so, we can discuss other alternatives.
> 
> Just my proposal. : )

Can you review it again?

File: tasks/task384_stance_detection_target_classification.json [hunk 7]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "#Mission:#Climate @ home > Buy drinking water & soda in big bottles... #Tip",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "@AlharbiF I'll bomb anything I can get my hands on, especially if THEY aren't christian. #graham2016 #GOP",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3280]: Atheism?

File: tasks/task384_stance_detection_target_classification.json [hunk 8]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "When debunking climate myths: fight sticky myths with stickier facts, mention the myth and explain the fallacy! ",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "I really can't stand the Catholic church but I think I have a crush on #PopeFrancis ! Shhhh, don't tell!  #AnimalRights",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3292]: Atheism?

File: tasks/task384_stance_detection_target_classification.json [hunk 9]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "@MarkRuffalo @flea333 @kellyslater unfortunate how USA refuses to take seriously damage we're responsible for #SaveTheEarth ",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Most people say that it is the intellect which makes a great scientist. They are wrong: it is character. ~ Albert Einstein ",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3310]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 10]
```diff
@@ -0,0 +1,16910 @@
...
+            ]
+        },
+        {
+            "input": "@CBCNews @nationalpost The human mind is incapable of dealing with that of which it has no experience ",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3316]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 11]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "RT @victoriaswatson: @WhiteHouse @POTUS What a wonderful exchange between two of the world's greatest men! Really touching. #",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "I love this Pope. I don't care what religion you are, this guy is awesome.",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3328]: Atheism?

File: tasks/task384_stance_detection_target_classification.json [hunk 12]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "RT @transitionlouth: @BP_plc We need degrowth - stop destroying our planet, BP.",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "@ParvaSaeua not to worry, our real enemy is #Obama",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3370]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 13]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "I don't want to be \"that guy\" but can we now make \"climate change denier marriage\" illegal? #MarriageEquaility #Idea",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Seriously, how many times can #Republicans be wrong? #ACA #LGBTRights #GovernmentShutdown #AutoBailout #StimulusSpending",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3454]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 14]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "@_A___L___I_ well they should! Why do you think people in Karachi r dying? because we don't care about our environment!",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Why is nothing being done about ",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3508]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 15]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "When the Last Tree Is Cut Down, the Last Fish Eaten & the Last Stream Poisoned, You Will Realize That You Cannot Eat Money",
+            "output": [
+                "Climate Change is a Real Concern"
+            ]
+        },
+        {
+            "input": "Australia is leading the way. Backwards. #MarriageEquaility #auspol",
+            "output": [
+                "Climate Change is a Real Concern"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:3544]: not related to climate change

File: tasks/task384_stance_detection_target_classification.json [hunk 16]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "Anyone find it funny that #Feminists use #TakeitLikeAMan when they don't want to understand what it means to be one? ",
+            "output": [
+                "Feminist Movement"
+            ]
+        },
+        {
+            "input": ".@MyRainbowNinja @NotAllBhas This farm is fucking disgusting.",
+            "output": [
+                "Feminist Movement"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:5734]: not related to feminist movement

File: tasks/task384_stance_detection_target_classification.json [hunk 17]
```diff
@@ -0,0 +1,16910 @@
...
+        {
+            "input": "Starting to compile evidence for blog post, how modern feminism shares the same traits as the Nazi propaganda #antifeminism",
+            "output": [
+                "Feminist Movement"
+            ]
+        },
+        {
+            "input": "Why don't you love me Twitter",
+            "output": [
+                "Feminist Movement"
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:5782]: not related to feminist

File: tasks/task384_stance_detection_target_classification.json [hunk 18]
```diff
@@ -0,0 +1,16910 @@
...
+            ]
+        },
+        {
+            "input": "Whether you label yourself a feminist or not I think it's important that we address equal rights.-Taylor Swift.",
+            "output": [
+                "Feminist Movement"
+            ]
+        },
+        {
+            "input": "#WeNeedFe
```
Comment (XudongOliverShen) [inline tasks/task384_stance_detection_target_classification.json:9892]: not related to Hillary Clinton

**Review by XudongOliverShen — COMMENTED** [review]
I have concerns over the noise in the dataset (see my annotations on task 384). Ideally, these bad instances should be removed. But it might require too large an effort from the contributor. any suggestions? @danyaljj 

another minor comment is you should shuffle the instances, as instructed here https://github.com/allenai/natural-instructions-expansion/pull/159#issuecomment-913248765

**Comment by swarooprm** [comment]
@cosmicishan, could you address @XudongOliverShen's comments?

**Comment by swarooprm** [comment]
@cosmicishan, a smaller and noise free dataset is better than a bigger one. Often we do that by leveraging inter-annotator agreements. If inter-annotation agreement is high, it typically implies annotations are correct. Check if you have access to these in the meta data.

**Comment by cosmicishan** [comment]
> @cosmicishan, a smaller and noise free dataset is better than a bigger one. Often we do that by leveraging inter-annotator agreements. If inter-annotation agreement is high, it typically implies annotations are correct. Check if you have access to these in the meta data.

I certainly don't have access to it. What do you suggest I should do? Should I close this pull request and replace it with another task from another dataset?

**Comment by danyaljj** [comment]
Yeah, if we can't improve the quality of the instances, unfortunately, we can't include this task.

**Comment by cosmicishan** [comment]
> Yeah, if we can't improve the quality of the instances, unfortunately, we can't include this task.

Should I close this pull request and add another task from another dataset and add these numbers?

**Comment by swarooprm** [comment]
> > Yeah, if we can't improve the quality of the instances, unfortunately, we can't include this task.
> 
> Should I close this pull request and add another task from another dataset and add these numbers?

yes, go ahead and close this PR. Alternatively if you think one of the tasks can have noise-free annotation, consider retaining that before you close the PR @cosmicishan
